<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1516</width>
    <height>935</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>USB Tool</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QListView" name="deviceList"/>
      <widget class="QTreeView" name="interfacesTreeView">
       <property name="headerHidden">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QStackedWidget" name="pageStack">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="descriptorPage">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="deviceDescriptorLabel">
           <property name="frameShape">
            <enum>QFrame::Panel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Sunken</enum>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="textFormat">
            <enum>Qt::RichText</enum>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="controlTransferPage">
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Control Transfer</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,1">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,0">
              <item>
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>bRequest:</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="PaddedSpinBox" name="bRequestSpinbox">
                  <property name="prefix">
                   <string>0x</string>
                  </property>
                  <property name="maximum">
                   <number>255</number>
                  </property>
                  <property name="displayIntegerBase">
                   <number>16</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_5">
                  <property name="text">
                   <string>wIndex:</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="PaddedSpinBox" name="wIndexSpinbox">
                  <property name="prefix">
                   <string>0x</string>
                  </property>
                  <property name="maximum">
                   <number>65535</number>
                  </property>
                  <property name="displayIntegerBase">
                   <number>16</number>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>wValue:</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="PaddedSpinBox" name="wValueSpinbox">
                  <property name="prefix">
                   <string>0x</string>
                  </property>
                  <property name="maximum">
                   <number>65535</number>
                  </property>
                  <property name="displayIntegerBase">
                   <number>16</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout">
                <item>
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>bmRequestType:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
                  <item>
                   <widget class="QLabel" name="label">
                    <property name="text">
                     <string>Recipient</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>Type</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_2">
                  <item>
                   <widget class="QListWidget" name="recipientList">
                    <property name="maximumSize">
                     <size>
                      <width>16777215</width>
                      <height>100</height>
                     </size>
                    </property>
                    <item>
                     <property name="text">
                      <string>Device</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>Interface</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>Endpoint</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>Other</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item>
                   <widget class="QListWidget" name="typeList">
                    <property name="maximumSize">
                     <size>
                      <width>16777215</width>
                      <height>100</height>
                     </size>
                    </property>
                    <item>
                     <property name="text">
                      <string>Standard</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>Class</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>Vendor</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QFormLayout" name="formLayout">
              <property name="fieldGrowthPolicy">
               <enum>QFormLayout::ExpandingFieldsGrow</enum>
              </property>
              <item row="0" column="0">
               <widget class="QPushButton" name="controlSendButton">
                <property name="text">
                 <string>Send (out)</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="controlSendDataEdit">
                <property name="placeholderText">
                 <string>Type hex here...</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="controlReceiveButton">
                <property name="text">
                 <string>Receive (in)</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <widget class="QLabel" name="label_7">
                  <property name="text">
                   <string>Requested Length:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="controlReceiveLengthSpinbox">
                  <property name="minimumSize">
                   <size>
                    <width>100</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="maximum">
                   <number>10000</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QSplitter" name="splitter_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <widget class="QLabel" name="controlReceiveDataLabelHex">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="palette">
                <palette>
                 <active>
                  <colorrole role="Base">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                  <colorrole role="Window">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                 </active>
                 <inactive>
                  <colorrole role="Base">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                  <colorrole role="Window">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                 </inactive>
                 <disabled>
                  <colorrole role="Base">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                  <colorrole role="Window">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                 </disabled>
                </palette>
               </property>
               <property name="frameShape">
                <enum>QFrame::Panel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Sunken</enum>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="textFormat">
                <enum>Qt::PlainText</enum>
               </property>
               <property name="alignment">
                <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
              <widget class="QLabel" name="controlReceiveDataLabelAscii">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="palette">
                <palette>
                 <active>
                  <colorrole role="Base">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                  <colorrole role="Window">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                 </active>
                 <inactive>
                  <colorrole role="Base">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                  <colorrole role="Window">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                 </inactive>
                 <disabled>
                  <colorrole role="Base">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                  <colorrole role="Window">
                   <brush brushstyle="SolidPattern">
                    <color alpha="255">
                     <red>255</red>
                     <green>255</green>
                     <blue>255</blue>
                    </color>
                   </brush>
                  </colorrole>
                 </disabled>
                </palette>
               </property>
               <property name="frameShape">
                <enum>QFrame::Panel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Sunken</enum>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="textFormat">
                <enum>Qt::PlainText</enum>
               </property>
               <property name="alignment">
                <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="bulkTransferPage">
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Bulk / Interrupt Transfer</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1516</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PaddedSpinBox</class>
   <extends>QSpinBox</extends>
   <header>PaddedSpinBox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>deviceList</tabstop>
  <tabstop>bRequestSpinbox</tabstop>
  <tabstop>wIndexSpinbox</tabstop>
  <tabstop>wValueSpinbox</tabstop>
  <tabstop>recipientList</tabstop>
  <tabstop>typeList</tabstop>
  <tabstop>controlSendButton</tabstop>
  <tabstop>controlSendDataEdit</tabstop>
  <tabstop>controlReceiveButton</tabstop>
  <tabstop>controlReceiveLengthSpinbox</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
